src/material-experimental/mdc-list/list-item-sections.ts

```diff
@@ -0,0 +1,100 @@
+ *
+ * Text content for the title never wraps. There can only be a single title per list item.
+ */
+@Directive({
+  selector: '[matListItemTitle]',
+  host: {'class': 'mat-mdc-list-item-title mdc-list-item__primary-text'},
+})
+export class MatListItemTitle {
+  constructor(public _elementRef: ElementRef<HTMLElement>) {}
+}
+
+/**
+ * Directive capturing a line in a list item. A list item usually consists of a
+ * title and optional secondary or tertiary lines.
+ *
+ * Text content inside a line never wraps. There can be at maximum two lines per list item.
+ */
+@Directive({
+  selector: '[matListItemLine]',
+  host: {'class': 'mat-mdc-list-item-line mdc-list-item__secondary-text'},
+})
+export class MatListItemLine {
+  constructor(public _elementRef: ElementRef<HTMLElement>) {}
+}
+
+/**
+ * Directive matching an optional meta section for list items.
+ *
+ * List items can reserve space at the end of an item to display a control,
+ * button or additional text content.
+ */
+@Directive({
+  selector: '[matListItemMeta]',
+  host: {'class': 'mat-mdc-list-item-meta mdc-list-item__end'},
+})
+export class MatListItemMeta {}
+
+/**
+ * @docs-private
+ *
+ * MDC uses the very intuitively named classes `.mdc-list-item__start` and `.mat-list-item__end`
+ * to position content such as icons or checkboxes that comes either before or after the text
+ * content respectively. This directive detects the placement of the checkbox and applies the
+ * correct MDC class to position the icon/avatar on the opposite side.
+ */
+@Directive({
+  host: {
+    // MDC uses intuitively named classes `.mdc-list-item__start` and `.mat-list-item__end`
+    // to position content such as icons or checkboxes that comes either before or after the text
+    // content respectively. This directive detects the placement of the checkbox and applies the
+    // correct MDC class to position the icon/avatar on the opposite side.
+    '[class.mdc-list-item__start]': '_isAlignedAtStart()',
+    '[class.mdc-list-item__end]': '!_isAlignedAtStart()',
+  },
+})
+export class _MatListItemGraphicBase {
+  constructor(@Optional() @Inject(LIST_OPTION) public _listOption: ListOption) {}
+
+  _isAlignedAtStart() {
+    // By default, in all list items the graphic is aligned at start. In list options,
+    // the graphic is only aligned at start if the checkbox is at the end.
+    return !this._listOption || this._listOption?._getCheckboxPosition() === 'after';
+  }
+}
+
+/**
+ * Directive matching an optional avatar within a list item.
+ *
+ * List items can reserve space at the beginning of an item to display an avatar.
+ */
+@Directive({
+  selector: '[matListItemAvatar]',
+  host: {'class': 'mat-mdc-list-item-avatar'},
+})
+export class MatListItemAvatar extends _MatListItemGraphicBase {}
+
+/**
+ * Directive matching an optional icon within a list item.
+ *
+ * List items can reserve space at the beginning of an item to display an icon.
+ */
+@Directive({
+  selector: '[matListItemIcon]',
```

Comment:
what about `[mat-list-icon],[matListIcon]`?

Reply:
I'm replying here for both the avatar and the icon comment:

Given that the class name has been renamed anyway from `matListIcon` to `matListItemIcon`, it seemed like we should just get rid of the dash-cases variants which seem like an old artifact given that our [standards](https://github.com/angular/components/blob/master/CODING_STANDARDS.md#selectors) prefer the camel-cased variant for directives. And since the migration is not a drop-in-replacement anyway, I was thinking of just getting rid of it from the beginning on (as we will need some advanced migration automation anyway)

Reply:
Given that our migration script is necessarily going to have to modify user's templates for this component, I'm ok with making this change.